sources/common/src/XmlSignalFactory.cpp

```diff
@@ -70,106 +70,104 @@ PARSER::XmlDescription::XmlDescription(const pugi::xml_node &rNode) :
 }
 
-const std::string & PARSER::XmlDescription::GetName() const
+const std::string &PARSER::XmlDescription::GetName() const
 {
     if (m_Name.length())
     {
         return m_Name;
     }
     throw EXCEPTION::VcdException(EXCEPTION::Error::NO_NAME_EXPRESSION,
-        "XML - No name expression.");
+                                  "XML - No name expression.");
 }
 
-const std::string & PARSER::XmlDescription::GetValue() const
+const std::string &PARSER::XmlDescription::GetValue() const
 {
     if (m_Value.length())
     {
         return m_Value;
     }
     throw EXCEPTION::VcdException(EXCEPTION::Error::NO_VALUE_EXPRESSION,
-        "XML - No value expression.");
+                                  "XML - No value expression.");
 }
 
-const std::string & PARSER::XmlDescription::GetSize() const
+const std::string &PARSER::XmlDescription::GetSize() const
 {
     if (m_Size.length())
     {
         return m_Size;
     }
     throw EXCEPTION::VcdException(EXCEPTION::Error::NO_SIZE_EXPRESSION,
-        "XML - No size expression.");
+                                  "XML - No size expression.");
 }
 
-PARSER::XmlSignalFactory::XmlSignalFactory(const std::string& rXmlFileName) : SignalFactory(false)
+PARSER::XmlSignalFactory::XmlSignalFactory(const std::string &rXmlFileName) :
+    SignalFactory()
 {
     // Parse the XML and add signal creators.
-    // pugi test
     pugi::xml_document doc;
-    if (!doc.load_file(rXmlFileName.c_str())) throw (1);  /// @todo Fix filename and error
+    if (!doc.load_file(rXmlFileName.c_str()))
+    {
+        throw EXCEPTION::VcdException(EXCEPTION::Error::XML_READ_ERROR,
+                                      "Opening XML file '" + rXmlFileName + "' failed. The XML file might be incorrect.");
+    }
 
     pugi::xml_node signals = doc.child("signals");
 
     // Traverse the XML.
-    for (pugi::xml_node signal : signals.children())
+    for (const pugi::xml_node &signal : signals.children())
     {
         if (!std::strcmp("vector", signal.name()))
         {
             std::unique_ptr<XmlDescription> description = std::make_unique<XmlDescription>(signal);
-            m_vpSignalCreators.push_back(std::make_unique<XmlISignalCreator>(
-                description->GetRegex(),
-                description->GetTimestamp(),
-                description->GetName(),
-                description->GetValue(),
-                description->GetSize()));
+            m_vpSignalCreators.push_back(std::make_unique<XmlISignalCreator>(description->GetRegex(),
+                                                                             description->GetTimestamp(),
+                                                                             description->GetName(),
+                                                                             description->GetValue(),
+                                                                             description->GetSize()));
             continue;
```

Comment:
Hmmm... Instead of `continue`  you could use `if () {} else if () {} ... else { throw; }`. I think it would be cleaner. (There are also other solutions for that, more pattern-like).